File: bao-source/goose-word-native-shell-bao/ios/App/App/Plugins/GooseWordServerPlugin.swift
import Foundation
import Capacitor

@objc(GooseWordServerPlugin)
public class GooseWordServerPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "GooseWordServerPlugin"
    public let jsName = "GooseWordServer"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "detect", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "start", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "stop", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getStatus", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "injectDeviceContext", returnType: CAPPluginReturnPromise),
    ]

    @objc func detect(_ call: CAPPluginCall) {
        call.resolve(["platform": "ios"])
    }

    @objc func start(_ call: CAPPluginCall) {
        let port = UInt16(call.getInt("port") ?? 8080)
        let dataDir = call.getString("dataDir") ?? NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true).first ?? ""
        let started = GooseWordLoopbackHost.shared.start(port: port, dataDir: dataDir + "/goose-word")
        call.resolve([
            "running": started,
            "loopbackUrl": "http://127.0.0.1:\(port)",
            "platform": "ios",
        ])
    }

    @objc func stop(_ call: CAPPluginCall) {
        GooseWordLoopbackHost.shared.stop()
        call.resolve(["stopped": true])
    }

    @objc func getStatus(_ call: CAPPluginCall) {
        let port = GooseWordLoopbackHost.shared.port
        call.resolve([
            "running": true,
            "loopbackUrl": "http://127.0.0.1:\(port)",
            "platform": "ios",
        ])
    }

    @objc func injectDeviceContext(_ call: CAPPluginCall) {
        call.resolve(["ok": true])
    }
}

File: bao-source/goose-word-native-shell-bao/android/app/src/main/java/com/baohaus/gooseword/GooseWordServerPlugin.java
package com.baohaus.gooseword;

import android.content.Intent;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "GooseWordServer")
public class GooseWordServerPlugin extends Plugin {
    @PluginMethod
    public void detect(PluginCall call) {
        JSObject result = new JSObject();
        result.put("platform", "android");
        call.resolve(result);
    }

    @PluginMethod
    public void start(PluginCall call) {
        int port = call.getInt("port", 8080);
        String dataDir = call.getString("dataDir", "");
        Intent intent = new Intent(getContext(), GooseWordForegroundService.class);
        intent.putExtra(GooseWordForegroundService.EXTRA_PORT, port);
        intent.putExtra(GooseWordForegroundService.EXTRA_DATA_DIR, dataDir);
        getContext().startForegroundService(intent);
        JSObject result = new JSObject();
        result.put("running", true);
        result.put("loopbackUrl", "http://127.0.0.1:" + port);
        result.put("platform", "android");
        call.resolve(result);
    }

    @PluginMethod
    public void stop(PluginCall call) {
        Intent intent = new Intent(getContext(), GooseWordForegroundService.class);
        getContext().stopService(intent);
        JSObject result = new JSObject();
        result.put("stopped", true);
        call.resolve(result);
    }

    @PluginMethod
    public void getStatus(PluginCall call) {
        JSObject result = new JSObject();
        result.put("running", true);
        result.put("loopbackUrl", "http://127.0.0.1:8080");
        result.put("platform", "android");
        call.resolve(result);
    }

    @PluginMethod
    public void injectDeviceContext(PluginCall call) {
        JSObject result = new JSObject();
        result.put("ok", true);
        call.resolve(result);
    }
}
